File: floating_window/k_elements_max_sum.js


function k_elements_max_sum(arr, k) {
    let max_sum = 0
    let current_sum = sum(arr.slice(0, k))
    max_sum = current_sum

    for (let right = k; right < arr.length; right++) {
        let left = right - k
        current_sum += arr[right] - arr[left]
        max_sum = Math.max(max_sum, current_sum)
    }
    
    return max_sum
}

console.log(k_elements_max_sum([1, 2, 3, 4, 5], 2))  // Output: 9 (4 + 5)

File: floating_window/k_elements_max_sum.py


def k_elements_max_sum(arr, k):
    max_sum = 0
    current_sum = sum(arr[:k])
    max_sum = current_sum

    for right in range(k, len(arr)):
        left = right - k
        current_sum += arr[right] - arr[left]
        max_sum = max(max_sum, current_sum)

    return max_sum

print(k_elements_max_sum([1, 2, 3, 4, 5], 2))  # Output: 9 (4 + 5)